### PR TITLE
gitignore file, stricter shebang matching, updated read me

### PR DESCRIPTION
**Pull request by homebysix**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.DS_Store
+
+# Virtual environment for Python 3
+venv/*
+
+# Compiled Python
+*.pyc
+
+# Example scripts and EAs
+extension_attributes/*
```
Comment (bradschm) [inline .gitignore:10]: @homebysix This still allows tracking of sub folders, correct? Is the idea that this line will keep errant files from being tracked within extension_attributes and scripts?

Comment (homebysix) [reply .gitignore:10]: Yes — line 10 excludes extension attributes, but line 11 re-includes the example EAs in the repo. Same for the scripts in lines 12-13.

This change is primarily for the benefit of those who want to contribute to the development of git2jss without having to manually remove downloaded scripts and extension attributes from the repo before committing/pushing.

Alternatively, I thought the ability to specify an alternate destination for the extension attributes and scripts would be a nice feature-add, but didn't have time yet to implement it.

**Comment by badstreff** [comment]
I like this for my personal dev purposes, it might cause confusion for people that are forking the project to their internal repos and then doing a `git push`. I think we will probably want to update the README to clarify if they want to push scripts to another repo they will need to edit the .gitignore.

**Comment by homebysix** [comment]
If people are forking the project to their internal repos, they'll already need to know how to handle multiple remotes in order to continue getting the latest changes from upstream.

Would it be better to add the ability to download extension attributes and scripts to a location outside the repo? If so, I can work on that.

**Comment by badstreff** [comment]
I can agree with that first statement, I think long term Brad and I have both tossed around the idea of specifying an alternative location for scripts but I don't think we need to hold this PR up for it.

**Pull request merged**